### PR TITLE
Use StringIO for generating event data for Stem

### PR DESCRIPTION
**Pull request by sammyshj**
Changes made in the PR:
* Uses StringIO.StringIO now in place of string concatenation for generating event data for Stem

### CODE REVIEW

File: txtorcon/torcontrolprotocol.py [hunk 1]
```diff
@@ -128,20 +129,18 @@ def got_update(self, data):
...
 
-        # XXX at the very least, use StringIO or something to avoid
-        # all the string-concatenations
-
+        response_data = StringIO()
         lines = data.split('\n')
         if len(lines) > 1:
-            response_data = "650-" + self.name + "\r\n"
+            response_data.write("650-" + self.name + "\r\n")
             for line in range(len(lines)-1):
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: looking good, but can you eliminate these +'s as well since we've got a stringio in here now?

Comment (sammyshj) [inline txtorcon/torcontrolprotocol.py]: Does this mean that we should have multiple `response_data.write()`?

Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: yes

Comment (sammyshj) [inline txtorcon/torcontrolprotocol.py]: Updated the PR

File: txtorcon/torcontrolprotocol.py [hunk 2]
```diff
@@ -128,20 +129,28 @@ def got_update(self, data):
...
-
+        response_data = StringIO()
         lines = data.split('\n')
         if len(lines) > 1:
-            response_data = "650-" + self.name + "\r\n"
+            response_data.write("650-")
+            response_data.write(self.name)
+            response_data.write("\r\n")
             for line in range(len(lines)-1):
-                response_data += "650-" + lines[line] + "\r\n"
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: Just "for line in lines" please

Comment (sammyshj) [inline txtorcon/torcontrolprotocol.py]: Updated!

**Pull request merged**